### PR TITLE
Update ``make`` module honor the return code of the ``make`` command

### PR DESCRIPTION
**Pull request by lpirl**
##### SUMMARY
tl;dr: ``make`` module now fails (as expected) if the ``make`` command returned a non-zero exit code.

Observed behavior: tasks using the ``make`` module did not fail despite the ``make`` command returned a non-zero exit code. This leads to a  – excuse me – terrible user experience when authoring Playbooks etc.

Behavior modified to: return code of ``make`` command is honored, so that tasks fail if the ``make`` command returned a non-zero exit code.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
``make`` module

##### ANSIBLE VERSION

```
  config file = /home/johndoe/.ansible.cfg
  configured module search path = [u'/home/johndoe/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.15+ (default, Aug 31 2018, 11:56:52) [GCC 8.2.0]
```

##### ADDITIONAL INFORMATION
Well, I am not sure I completely get what the comment right above my change is trying to tell me.
However, if the ``make`` module does not honor the return code of the ``make`` command, users are better off without this module, since it makes it difficult to identify sources of unexpected behavior.
E.g.: you wonder why your remote setup does not work; you spend a lot of time on debugging, just to find out that your Makefile did not run (which Ansible did not tell you) because of a missing dependency or something likewise trivial.

Hope this patch is adequate.

### CODE REVIEW

**Comment by Shaps** [comment]
@lpirl I think what the comment means is that the RC of the make is checked within `run_command` which will in turn fail when/if the rc is `!= 0 `. 

That said, `check_rc` in `run_command` is set to `False` by default [basic.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L2706), and `run_command` will only fail when it's set to `True`. I guess the right solution would be to set `check_rc=True` here: [make.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/system/make.py#L143)

**Comment by gundalow** [comment]
@lpirl Hi, thanks for the bug fix, does the comment above answer your question?

**Comment by lpirl** [comment]
Thanks for the hint @Shaps and the reminder @gundalow.
With the accordingly updated changes, I get the desired result as well (i.e., module fails if ``make``'s return code is not ``0``).

**Comment by Shaps** [comment]
shipit

**Comment by gundalow** [comment]
Merged into `devel` for release in Ansible 2.8, thanks!

If you'd like this fixed in 2.7 please raise a backport PR including a changelog https://docs.ansible.com/ansible/devel/community/development_process.html#backport-pull-request-process

**Comment by lpirl** [comment]
Sweet, cheers everyone.
